File: Selenium/Project/Activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import time
from selenium.webdriver import ActionChains

with webdriver.Firefox() as driver:
    driver.get("http://alchemy.hguy.co/crm")
    driver.maximize_window()
    action = ActionChains(driver)
    wait = WebDriverWait(driver, 10)

    user_name_input = driver.find_element(By.ID, "user_name")
    password_input = driver.find_element(By.ID, "username_password")
    login_button = driver.find_element(By.ID, "bigbutton")
    user_name_input.send_keys("admin")
    password_input.send_keys("pa$$w0rd")
    login_button.click()
    time.sleep(5)
    actions_button_home_page = driver.find_element(By.ID, "tab-actions")
    wait.until(EC.visibility_of(actions_button_home_page))

    sales_menu_item = driver.find_element(By.XPATH, "//a[text() = 'Sales']")
    accounts = driver.find_element(By.XPATH, "//a[text() = 'Accounts']")
    action.move_to_element(sales_menu_item).pause(3).click(accounts).perform()

    wait.until(EC.visibility_of_element_located((By.CSS_SELECTOR, ".table-responsive")))
    time.sleep(5)

    for i in range(1,10,2):
        rows_xpath_text = f"//div[@class= 'list-view-rounded-corners']/table/tbody/tr[{i}]"
        row = driver.find_elements(By.XPATH, rows_xpath_text)
        for col in row:
            col = driver.find_element(By.XPATH, f"{rows_xpath_text}/td[@field = 'name']//a")
            print(f"Name of account {i}: {col.text}")

File: Selenium/Project/Activity8.java
package project;

import java.time.Duration;
import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Reporter;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class Activity8 {
	public WebDriver driver;
	public WebDriverWait wait;

	@BeforeClass
	public void setup() {
		driver = new FirefoxDriver();
		wait = new WebDriverWait(driver, Duration.ofSeconds(10));
		driver.get("http://alchemy.hguy.co/crm");
		driver.manage().window().maximize();
	}

	@Test
	public void testActivity8() throws InterruptedException {
		WebElement userNameInput = driver.findElement(By.id("user_name"));
		WebElement passwordInput = driver.findElement(By.id("username_password"));
		WebElement loginButton = driver.findElement(By.id("bigbutton"));
		userNameInput.sendKeys("admin");
		passwordInput.sendKeys("pa$$w0rd");
		loginButton.click();
		Thread.sleep(Duration.ofSeconds(5));
		WebElement actionsButtonHomePage = driver.findElement(By.id("tab-actions"));
		wait.until(ExpectedConditions.visibilityOf(actionsButtonHomePage));

		Actions action = new Actions(driver);
		WebElement salesMenuItem = driver.findElement(By.xpath("//a[text() = 'Sales']"));
		WebElement accounts = driver.findElement(By.xpath("//a[text() = 'Accounts']"));
		action.moveToElement(salesMenuItem).pause(Duration.ofSeconds(3)).click(accounts).pause(Duration.ofSeconds(5)).build().perform();

		wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".table-responsive")));
		Thread.sleep(Duration.ofSeconds(3));

		String rowsXpathText;
		List<WebElement> row;
		for(int i = 1; i <= 10; i+=2) {
			rowsXpathText = "//div[@class= 'list-view-rounded-corners']/table/tbody/tr["+i+"]";
			row = driver.findElements(By.xpath(rowsXpathText));

			for (WebElement col : row) {
				col = driver.findElement(By.xpath(rowsXpathText+"/td[@field = 'name']//a"));

				System.out.println("Name of account "+i+": "+col.getText());
				Reporter.log("Name of account "+i+": "+col.getText());
			}
		}
	}
	@AfterClass
	public void teardown() {
		driver.quit();
	}
}
